update.ps1

```diff
@@ -0,0 +1,98 @@
+#region variables
+
+$script:rancherDesktopExe = "C:\Users\$env:UserName\AppData\Local\Programs\Rancher Desktop\Rancher Desktop.exe"
+$script:rancherDesktopExeHash = "CF7E00240316A3654AB66802A8AAA281478824650C4032C1862123C317CF0885"
+$script:rancherDesktopVersion = "1.1.1"
+$script:windowsBinariesPath = "C:\Users\$env:UserName\AppData\Local\Programs\Rancher Desktop\resources\resources\win32\bin"
+$script:linuxBinariesPath = "C:\Users\$env:UserName\AppData\Local\Programs\Rancher Desktop\resources\resources\linux\bin"
+$script:rancherDesktopTargetVersion = "1.3.0"
+$script:rancherDesktopInstallerName = "Rancher.Desktop.Setup.$script:rancherDesktopTargetVersion"
+$script:rancherDesktopUrl = "https://github.com/rancher-sandbox/rancher-desktop/releases/download/v$script:rancherDesktopTargetVersion/$script:rancherDesktopInstallerName.exe"
+$script:rancherDesktopInstallerHash = "92108CBBD8C98F99B00A608D8F7D21E12FAECA76F16890585EF212CC5BF1C779"
+
+#endregion
+
+#region functions
+
+function RenameBinariesFunction
+{
+    Write-Host "Renaming the Rancher Desktop binaries..." -ForegroundColor Blue
+    Rename-Item -Path "$script:windowsBinariesPath\docker.exe" -NewName dockerw.exe
+    Rename-Item -Path "$script:windowsBinariesPath\docker-compose.exe" -NewName dockerw-compose.exe
+    Copy-Item "$script:windowsBinariesPath\nerdctl.exe" "$script:windowsBinariesPath\docker.exe" -Force
+
+    Rename-Item -Path "$script:linuxBinariesPath\docker" -NewName docker.old
+    Copy-Item "$script:linuxBinariesPath\nerdctl" "$script:linuxBinariesPath\docker" -Force
+    Write-Host "Renaming done." -ForegroundColor Green
+}
+
+function UpdateRancherDesktop
+{
+
+    Write-Host "Updating Rancher Desktop to version $script:rancherDesktopTargetVersion..." -ForegroundColor Blue
+    
+    if(!(Test-Path -Path $script:rancherDesktopExe))
+    {
+        Write-Host "Rancher Desktop was not found in the system." -ForegroundColor Red
+        exit 1
+    }
+    elseif ((Get-FileHash -Algorithm SHA256 "$script:rancherDesktopExe").Hash -ne "$script:rancherDesktopExeHash")
+    {
+        Write-Host "Wrong Rancher Desktop version detected. Please make sure that the version installed is: $script:rancherDesktopVersion" -ForegroundColor Red
+        exit 2      
+    }
+    else # All update preconditions passed
+    {
+        # Create backup of dockerd
+        Copy-Item "$script:windowsBinariesPath\dockerd.exe" "C:\" -Recurse -Force
```

Comment:
Better create a temporary file (https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/new-temporaryfile?view=powershell-7.2) and copy the file there.

Please apply also that change to installation to avoid writing directly on C: when decompressing dockerd